### PR TITLE
Bugfix/warning messages with g option

### PR DESCRIPTION
**Pull request by b-carr**
This removes the warnings that used to appear if you used ```-g``` without an output file. It also makes it possible to use ```-g``` without an input file.

### CODE REVIEW

File: src/main/seal.c [hunk 1]
```diff
@@ -266,13 +266,21 @@ int main(int argc, char **argv)
     }
   }
 
+  // Some options don't do anything with -g, so warn about that now.
+  if(boolTrialOnly)
+  {
+    if(outPath != NULL || inPath != NULL || cipherString != NULL || forceOverwrite || expected_policy != NULL)
```
Comment (bit-bender) [inline src/main/seal.c:272]:
```suggestion
    if(authString != NULL || ownerAuthPasswd != NULL || outPath != NULL || inPath != NULL || cipherString != NULL || forceOverwrite || expected_policy != NULL)
```

**Review by bit-bender — COMMENTED** [review]
Although this was not addressed by this pull request, I will only mention that it really bugs me that the actual code that prints the policy digest value does not include a newline character after the policy digest value. Since this type of minor "clean-up" is consistent with this pull request, I am suggesting that we also sneak a small change in to fix this.

I am much more ambivalent about the fact that the policy digest is not printed with any type of label.

In summary, on line 174 of 'src/tpm/kmyth_seal_unseal_impl.c', I would have coded it something like:

```
printf("policy digest: %s\n", output_string);
```

I realize that this is a very subjective and somewhat trivial comment, though, and am open to disregarding it or addressing it in a better way than I suggest.

File: include/kmyth.h [hunk 2]
```diff
@@ -56,6 +57,10 @@ extern "C"
  *                               policy digest to be used for authorizing
  *                               actions
  *
+ * @param[in]  bool_trail_only   Option indicating kmyth should only compute
```
[the remark this answers is not in the document]
Comment (b-carr) [reply include/kmyth.h]: Done in 1c61657

File: include/kmyth.h [hunk 3]
```diff
@@ -139,14 +144,19 @@ extern "C"
...
```
[the remark this answers is not in the document]
Comment (b-carr) [reply include/kmyth.h]: Done in 1c61657

**Comment by b-carr** [comment]
> Although this was not addressed by this pull request, I will only mention that it really bugs me that the actual code that prints the policy digest value does not include a newline character after the policy digest value. Since this type of minor "clean-up" is consistent with this pull request, I am suggesting that we also sneak a small change in to fix this.
> 
> I am much more ambivalent about the fact that the policy digest is not printed with any type of label.
> 
> In summary, on line 174 of 'src/tpm/kmyth_seal_unseal_impl.c', I would have coded it something like:
> 
> ```
> printf("policy digest: %s\n", output_string);
> ```
> 
> I realize that this is a very subjective and somewhat trivial comment, though, and am open to disregarding it or addressing it in a better way than I suggest.

Made this change in 1c61657

File: src/main/seal.c [hunk 4]
```diff
@@ -266,13 +266,21 @@ int main(int argc, char **argv)
     }
   }
 
+  // Some options don't do anything with -g, so warn about that now.
```
[the remark this answers is not in the document]
Comment (b-carr) [reply src/main/seal.c:269]: Updated to reflect the various options that don't impact ```-g``` in b273ce9 and da5d98d

File: src/main/seal.c [hunk 5]
```diff
@@ -269,9 +269,9 @@ int main(int argc, char **argv)
   // Some options don't do anything with -g, so warn about that now.
   if(boolTrialOnly)
   {
-    if(outPath != NULL || inPath != NULL || cipherString != NULL || forceOverwrite || expected_policy != NULL)
+    if(authString != NULL || ownerAuthPasswd != NULL || outPath != NULL || inPath != NULL || cipherString != NULL || forceOverwrite || expected_policy != NULL)
```
Comment (bit-bender) [inline src/main/seal.c:272]: When I tested, I got the warning message no matter what. In looking at it, I think it was because `ownerAuthPasswd` is initialized to the empty string (""), not NULL. When I changed the criteria to `ownerAuthPasswd != ""`, therefore, this corrected it.

```suggestion
    if(authString != NULL || ownerAuthPasswd != "" || outPath != NULL || inPath != NULL || cipherString != NULL || forceOverwrite || expected_policy != NULL)
```
